pkgs/top-level/haskell-packages.nix

```diff
@@ -115,68 +118,78 @@ in rec {
   # Default overrides that are applied to all package sets.
   packageOverrides = self : super : {};
 
-  # Always get compilers from `buildPackages`
```

Comment:
Probably I shouldn’t have removed this part.
But I have to admit that I don’t understand the difference between a compiler from `packages` and a compiler from `buildPackages`. I think one from `packages` is expected to run on host and target target, while the one from `buildPackages` is expected to run on build and target host, is that right? In that case, how do we a get a compiler used for `Setup.hs`, i.e. one that both runs on and targets build?
@Ericson2314